lib/StripeResource.js

```diff
@@ -192,14 +192,53 @@ StripeResource.prototype = {
       callback.call(
         self,
         new Error.StripeConnectionError({
-          message: 'An error occurred with our connection to Stripe',
+          message: 'An error occurred with our connection to Stripe. Request was retried ' + requestRetries + ' times.',
           detail: error,
         }),
         null
       );
     }
   },
 
+  _shouldRetry: function(res, numRetries) {
+    if (numRetries >= this._stripe.getMaxNetworkRetries()) {
+      return false;
+    }
+
+    // conflict or rate limiting
+    if (res && (res.statusCode === 409 || res.statusCode === 429)) {
+      return false;
+    }
+
+    // don't retry if everything succeeded
+    if (res && res.statusCode === 200) {
+      return false;
+    }
+
```

Comment:
Does this mean we retry 400 and 500 errors too? We should not, a 400 is a bad parameter error or similar and retrying won't help (+ it will also add delay). I think we should whitelist what makes a retry instead of always retry unless a specific behaviour happened